From 414fc605a930c5baa7e65cc303b05f3977b86baf Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Tue, 9 Jun 2015 23:29:40 -0700
Subject: [PATCH 2/8] XQuartz: Silence -Wpointer-bool-conversion

X11Controller.m:417:17: error: address of function 'asl_log_descriptor' will always evaluate to 'true'
      [-Werror,-Wpointer-bool-conversion,Value Conversion Issue]
            if (asl_log_descriptor) {
            ~~  ^~~~~~~~~~~~~~~~~~
X11Controller.m:417:17: note: prefix with the address-of operator to silence this warning [Semantic Issue]
            if (asl_log_descriptor) {
                ^
                &

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
(cherry picked from commit 0b9c32489131a5723bd78decf5d2557b94207cf4)
---
 hw/xquartz/X11Controller.m | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/hw/xquartz/X11Controller.m b/hw/xquartz/X11Controller.m
index 022e832..c3c6976 100644
--- a/hw/xquartz/X11Controller.m
+++ b/hw/xquartz/X11Controller.m
@@ -364,7 +364,7 @@ extern char *bundle_id_prefix;
     }
 
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
-    if (asl_log_descriptor) {
+    if (&asl_log_descriptor) {
         char *asl_sender;
         aslmsg amsg = asl_new(ASL_TYPE_MSG);
         assert(amsg);
@@ -414,7 +414,7 @@ extern char *bundle_id_prefix;
 
         case 0:                                     /* child2 */
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
-            if (asl_log_descriptor) {
+            if (&asl_log_descriptor) {
                 /* Replace our stdout/stderr */
                 dup2(stdout_pipe[1], STDOUT_FILENO);
                 dup2(stderr_pipe[1], STDERR_FILENO);
@@ -443,7 +443,7 @@ extern char *bundle_id_prefix;
     }
 
 #if MAC_OS_X_VERSION_MAX_ALLOWED >= 1080
-    if (asl_log_descriptor) {
+    if (&asl_log_descriptor) {
         /* Close the write ends of the pipe */
         close(stdout_pipe[1]);
         close(stderr_pipe[1]);
-- 
2.6.1

